boa3_test/tests/compiler_tests/test_range.py

```diff
@@ -134,28 +122,18 @@ def test_range_given_start(self):
             + Opcode.RET        # return
         )
 
-        path = self.get_contract_path('RangeGivenStart.py')
-        output = self.compile(path)
+        output, _ = self.assertCompile('RangeGivenStart.py')
         self.assertEqual(expected_output, output)
 
-        path, _ = self.get_deploy_file_paths(path)
-        runner = BoaTestRunner(runner_id=self.method_name())
-
-        invokes = []
-        expected_results = []
-
-        invokes.append(runner.call_contract(path, 'range_example', 2, 6))
-        expected_results.append([2, 3, 4, 5])
-        invokes.append(runner.call_contract(path, 'range_example', -10, 0))
-        expected_results.append([-10, -9, -8, -7, -6, -5, -4, -3, -2, -1])
-
-        runner.execute()
-        self.assertEqual(VMState.HALT, runner.vm_state, msg=runner.error)
+    async def test_range_given_start(self):
+        await self.set_up_contract('RangeGivenStart.py')
 
-        for x in range(len(invokes)):
-            self.assertEqual(expected_results[x], invokes[x].result)
+        result, _ = await self.call('range_example', [2, 6], return_type=list)
+        self.assertEqual([2, 3, 4, 5], result)
+        result, _ = await self.call('range_example', [-10, 0], return_type=list)
+        self.assertEqual([-10, -9, -8, -7, -6, -5, -4, -3, -2, -1], result)
```

Comment:
I know this is how the tests were before, but since we are refactoring this test case anyway, could you change the expected results to `list(range(args))`?
```suggestion
        result, _ = await self.call('range_example', [2, 6], return_type=list)
        self.assertEqual(list(range(2, 6)), result)
        result, _ = await self.call('range_example', [-10, 0], return_type=list)
        self.assertEqual(list(range(-10, 0)), result)
```